### PR TITLE
Refactor FXIOS-6477 [v121] Use Telemetry Wrapper in SearchViewController

### PR DESCRIPTION
**Pull request by sooraav**
…er, testcase written

## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-6477)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/14562)

## :bulb: Description

Used TelemetryWrapper and removed `import Glean` from SearchViewController

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [x] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: Client/Telemetry/TelemetryWrapper.swift [hunk 1]
```diff
@@ -1063,6 +1066,11 @@ extension TelemetryWrapper {
         // MARK: Start Search Button
         case (.action, .tap, .startSearchButton, _, _):
             GleanMetrics.Search.startSearchPressed.add()
+        case(.action, .tap, .recordSearch, _, let extras):
+            if let searchLocation = extras?[EventExtraKey.recordSearchLocation.rawValue] as? SearchesMeasurement.SearchLocation, let searchEngineID = extras?[EventExtraKey.recordSearchEngineID.rawValue] as? String? {
+                Telemetry.default.recordSearch(location: searchLocation, searchEngine: searchEngineID ?? "other")
+                GleanMetrics.Search.counts["\(searchEngineID ?? "custom").\(searchLocation.rawValue)"].add()
+            }
```
Comment (lmarceau) [inline Client/Telemetry/TelemetryWrapper.swift:1073]: Can we add an `else` case for `recordUninstrumentedMetrics` like we do in other places? This help us catch if we ever have a telemetry problem in this class.

**Review by lmarceau — COMMENTED** [review]
Great work! I have one small comment only

File: Client/Telemetry/TelemetryWrapper.swift (hunk 1 above)
Comment (sooraav) [reply Client/Telemetry/TelemetryWrapper.swift:1080]: Hi @lmarceau I have added `recordUninstrumentedMetrics `

**Review by lmarceau — APPROVED** [review]
Great work @sooraav !

**Pull request merged**